### PR TITLE
Test stability fixes

### PR DESCRIPTION
**Pull request by sharwell**
This pull request contains all the features from #27764 except for automatic retry for UI Automation calls. The documentation in **TextViewWindow_InProc.cs** was updated per the review from @tannergooding.

* Improves ability for integration test failures to be isolated to the particular test(s) that failed (785834f9787dcac564acfd801bdc3c0cfe0dc7c6)
* Apply additional timeouts to mitigate hangs (6476450665c4c983bbd23784a46402b4966ee2fe)
* Improve robustness of new .editorconfig Fix All tests (3e4cf945dbe3ef70e43d7d472c1276bcad1e1be8)

<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

**Comment by sharwell** [comment]
:memo: The failure in the es build is fixed by a separate pull request, #27843

**Comment by sharwell** [comment]
@jinujoseph for approval (there is one non-test file altered by this pull request, but only for the purpose of allowing tests to modify behavior so they don't hang).

**Comment by jinujoseph** [comment]
@Pilchie for Approval (since this is going into preview3)

**Comment by jinujoseph** [comment]
@jasonmalinowski  to review as well

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReplIdeFeatures.cs [hunk 1]
```diff
@@ -24,14 +24,19 @@ public CSharpReplIdeFeatures(VisualStudioInstanceFactory instanceFactory)
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing)
+            try
             {
-                VisualStudio.Workspace.SetUseSuggestionMode(false);
-                VisualStudio.InteractiveWindow.ClearReplText();
-                VisualStudio.InteractiveWindow.Reset();
+                if (disposing && VisualStudio != null)
```
Comment (jasonmalinowski) [inline src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReplIdeFeatures.cs]: Should we have a separate method to override (Cleanup() or the like) that does this pattern automatically?

Comment (sharwell) [reply src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReplIdeFeatures.cs]: 😆 So, it turns out @333fred called this ages ago (https://github.com/dotnet/roslyn/pull/25467#discussion_r174545764). `DisposeAsync` is only called if `InitializeAsync` completes successfully, so these methods should be paired for resource management, and the constructor initialization should be paired with `Dispose`.

**Comment by sharwell** [comment]
@jinujoseph I reverted the changes to production code in this pull request. I will file a follow-up pull request to restore the behavior that is currently pending shiproom approvals.

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReplIdeFeatures.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReplIdeFeatures.cs]: For some reason a bunch of tests failed when I made the switch in this PR, so I'm going to put it in the follow-up PR instead.

**Comment by sharwell** [comment]
Merging this test-only PR. The pull request *only* touches the integration tests, and those aren't an input to the **windows_debug_spanish_unit32_prtest** build completing successfully. Both of the integration test builds did complete successfully.

**Pull request merged**